common4j/src/main/com/microsoft/identity/common/java/opentelemetry/CertBasedAuthTelemetryHelper.java

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.opentelemetry;
+
+import javax.annotation.Nullable;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import lombok.NonNull;
+
+/**
+ * Assists classes associated with Certificate Based Authentication (CBA) with
+ *  telemetry-related tasks.
+ */
+public class CertBasedAuthTelemetryHelper {
+
+    /**
+     * Creates a new Span and sets it to be the current Span.
+     */
+    public static void createSpanAndMakeCurrent() {
+        final Span span = OTelUtility.createSpan(SpanName.CertBasedAuth.name());
+        span.makeCurrent();
```

Comment:
The scopes created by calls to `makeCurrent` must be closed at the end of those Spans. See this: https://github.com/open-telemetry/opentelemetry-java/blob/main/context/src/main/java/io/opentelemetry/context/ImplicitContextKeyed.java#L31

This is the reason Otel Java recommends using this in conjunction with `try-with-resources`: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Otherwise this is going to lead to memory leaks.

Reply:
A static variable (mScope) was created to hold the current Scope once span.makeCurrent() is called. 
mScope.close() is called within a null check in all of the setResult methods.